src/coreclr/jit/valuenum.cpp

```diff
@@ -3246,87 +3386,101 @@ bool ValueNumStore::VNEvalShouldFold(var_types typ, VNFunc func, ValueNum arg0VN
 ValueNum ValueNumStore::EvalUsingMathIdentity(var_types typ, VNFunc func, ValueNum arg0VN, ValueNum arg1VN)
 {
     ValueNum resultVN = NoVN; // set default result to unsuccessful
+    ValueNum ZeroVN;
```

Comment:
Hmm, I think they are still used, e. g.:
https://github.com/dotnet/runtime/blob/99a982999e806188b71db2f4dff62c4818559e7d/src/coreclr/jit/valuenum.cpp#L3493
https://github.com/dotnet/runtime/blob/99a982999e806188b71db2f4dff62c4818559e7d/src/coreclr/jit/valuenum.cpp#L3505
https://github.com/dotnet/runtime/blob/99a982999e806188b71db2f4dff62c4818559e7d/src/coreclr/jit/valuenum.cpp#L3519

A bit of a separate question is why are they not `camelCase`d and if it is worth it to micro-optimize by only requesting the values on the relevant code paths and otherwise leaving the locals in an uninitialized state (I presume it is because this method is actually called for each binary operation in the tree being numbered).

Reply:
I see, so we are shadowing them in lambdas, maybe move the declaration of `OneVN ` and `ZeroVN ` inside https://github.com/dotnet/runtime/blob/99a982999e806188b71db2f4dff62c4818559e7d/src/coreclr/jit/valuenum.cpp#L3470
?

Reply:
Agreed and fixed.